editor/utils/editor-media-upload/index.js

```diff
@@ -14,7 +14,7 @@ import { mediaUpload } from '@wordpress/utils';
  * Wrapper around mediaUpload() that injects the current post ID.
  *
  * @param   {Object}   $0                   Parameters object passed to the function.
- * @param   {string}   $0.allowedType       The type of media that can be uploaded.
+ * @param   {string}   $0.allowedType       The type of media that can be uploaded, or '*' to allow all.
```

Comment:
Is it better to require an explicit `'*'` or to make `allowedType` optional and assume no restriction? (I'm not sure)

Reply:
I had a think about it, and I think I prefer the `*` over making it optional. It at least signals a clear intention and is an accepted wildcard.

Making it optional and leaving it undefined could indicate either any or no allowed types.

Another option is to use a constant for the accepted values - e.g `allowedType: fileTypes.image` or `allowedType: fileTypes.any`.